docs/concepts/scale-up.md

```diff
@@ -0,0 +1,44 @@
+## Scaling-up the single-node to multi-node etcd cluster deployed by etcd-druid
+
+To mark the cluster for scale-up from single node to multi-node etcd just patch the etcd custom resource's `.spec.replicas` from `1` to `3`(say).
+
+### Challenges for scale-up
+1. Etcd cluster with single replica don't have any peer, so no peer communication is required hence peerUrl may or may not be TLS enabled but while scaling up from single node etcd to multi-node etcd cluster there will be a requirement to have a peer communication between each members of etcd cluster. Peer communication is required for various reasons like to get in a sync with each other cluster state, data and to perform leader election or any cluster wide operation like to remove/add a member etc. Hence in multi-node etcd cluster we need to have TLS enable peer url for peer communication.
+2. Providing the correct configuration to start a new etcd members as it is different from boostrapping a cluster since this new etcd members will join the existing cluster.
+
+### Approach
+We first went through the etcd doc of [update-advertise-peer-urls](https://etcd.io/docs/v3.4/op-guide/runtime-configuration/#update-advertise-peer-urls) to find out information regarding to update peerUrl. Interestingly etcd doc have mentioned this:
+```
+To update the advertise peer URLs of a member, first update it explicitly via member command and then restart the member.
+```
+
+But we can't assume peerUrl is not TLS enabled for single-node cluster as it depends on end-user, user may or may not enable the TLS for peerUrl for single node etcd cluster. so, How to detect whether peerUrl was enabled or not when cluster is marked for scale-up ?
+
+### Detection of peerUrl TLS enabled or not
+For this we have used member lease annotation `member.etcd.gardener.cloud/tls-enabled` set by backup-restore sidecar of etcd. As etcd configuration is provided by backup-restore, so backup-restore can find out whether TLS is enabled or not and accordingly set this annotation `member.etcd.gardener.cloud/tls-enabled: "true"` or `member.etcd.gardener.cloud/tls-enabled: "false"` in member lease object.
+And with the help of this annotation and config-map values etcd-druid is able to detect whether there is a change in a peerUrl or not.
+
+
+### Etcd-Druid helps in scaling up etcd cluster
+Now, we have detected whether peerUrl was TLS enabled or not for single node etcd cluster. Now, etcd-druid will use this information to take action:
+- If peerUrl was already TLS enabled then no action is required from etcd-druid side, etcd-druid can proceed for scaling up the cluster.
+- If peerUrl was not TLS enabled then etcd-druid have to intervene to make sure peerUrl should be TLS enabled first for the single node before marking the cluster for scale-up.
```

Comment:
```suggestion
- If peerURL was not TLS enabled then etcd-druid has to intervene and make sure that peerURL is first TLS enabled for the single node before marking the cluster for scale-up.
```

Please change all further occurrences of `peerUrl` to `peerURL`

Reply:
> Please change all further occurrences of peerUrl to peerURL

IMO, peerUrl seems correct as it follows camelCase.

Reply:
`URL` is an abbreviation and usually should be capitalised, but I'll leave it up to you

Reply:
make sense. done